### PR TITLE
Space - Stephanie 

### PR DESCRIPTION
**Pull request by stephaniejmars**
# Task List
Congratulations! You're submitting your assignment!

## Comprehension Questions
Question | Answer
:------------- | :-------------
Describe in your own words what the Model is doing in Rails | Model is the format for adding things to the DB. 
Describe in your own words what the Controller is doing in Rails | Controller is holding all the methods that get called with the routes and buttons 
Describe in your own words what the View is doing in Rails | Views are the code that is rednering on the page 
Describe an edge-case controller test you wrote |
What is the purpose of using strong params? (i.e. the params method in the controller) | 
How are Rails migrations related to Rails models? | Whenever you create a model or change a model you update a migration 
Describe one area of Rails that are still unclear on | Not sure why I had to make a path for mark_complete when I was not rendering a new page.

### CODE REVIEW

File: config/routes.rb [hunk 1]
```diff
@@ -0,0 +1,19 @@
+Rails.application.routes.draw do
+  # For details on the DSL available within this file, see https://guides.rubyonrails.org/routing.html
+
```
Comment (beccaelenzil) [inline config/routes.rb:3]: Consider adding a root path

File: app/views/tasks/show.html.erb [hunk 2]
```diff
@@ -0,0 +1,12 @@
+<h2><%= @task[:name]%></h2>
+<h4><%= @task[:description]%></h4>
```
Comment (beccaelenzil) [inline app/views/tasks/show.html.erb:2]: consider using dot notation. `@task.description`

**Comment by beccaelenzil** [comment]
# Task List




## Major Learning Goals/Code Review





| Criteria | yes/no, and optionally any details/lines of code to reference |
| --- | --- |
| At least 6 commits with meaningful commit messages | ✔️
| Routes follow RESTful conventions | ✔️
| Uses named routes (like `_path`) | ✔️
| Creates Models and migrations | ✔️
| Creates styled views | No, but not required
| Handles errors like nonexistant tasks | ✔️
| Uses `form_with` to render forms in Rails | ✔️

## Functional Requirements/Manual Testing

| Functional Requirement | yes/no |
| --- | --- |
| Successfully handles index & show | ✔️
| index & show tests pass | ✔️
| Successfully handles: New, Create | ✔️
| New, Create tests pass | Expected response to be a redirect to <http://www.example.com/tasks/2> but was a redirect to <http://www.example.com/tasks>.
        Expected "http://www.example.com/tasks/2" to be === "http://www.example.com/tasks".
| Successfully handles: Edit, Update | ✔️
| Successfully handles: Destroy, Task Complete | ✔️

## Overall Feedback

| Overall Feedback | Criteria | yes/no |
| --- | --- | --- |
| Green (Meets/Exceeds Standards) | 5+ in Code Review && 5+ in Functional Requirements |
| Yellow (Approaches Standards) | 3+ in Code Review && 4+ in Functional Requirements, or the instructor judges that this project needs special attention |
| Red (Not at Standard) | 0-2 in Code Review or 0-3 in Functional Reqs, or assignment is breaking/doesn’t run with less than 5 minutes of debugging, or the instructor judges that this project needs special attention |





## Code Style Bonus Awards



Was the code particularly impressive in code style for any of these reasons (or more...?)

| Quality | Yes? |
| --- | --- |
| Descriptive/Readable | ✅
| Concise | ✅
| Logical/Organized | ✅

**Review by beccaelenzil — COMMENTED** [review]
Good work on your first rails project. You've implemented the MVC paradigm in rails! It looks like you overwrote the provided tests when you created the project. When I ran the tests, one had an error (no root_path), and one failed because you redirected somewhere else. These are minor issues, but in the future make sure you get the provided tests for key information on requirements. Keep up the hard work!
